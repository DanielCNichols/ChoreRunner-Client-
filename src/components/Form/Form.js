import React from 'react';
import cx from 'classnames';
import s from './Form.module.css';

export function Label({ className, ...props }) {
  return <label className={cx(`${s.Label}`, className)} {...props} />;
}
export function Fieldset({ className, ...props }) {
  return (
    <fieldset className={cx(`${s.Fieldset}`, className)} {...props}></fieldset>
  );
}
export function Legend({ className, ...props }) {
  return <legend className={cx(`${s.Legend}`, className)} {...props}></legend>;
}

export const Input = React.forwardRef(({ className, ...props }, ref) => {
  return (
    <input
      className={cx(`${s.Input}`, className)}
      type="text"
      ref={ref}
      {...props}
    />
  );
});

export function Required({ className, ...props }) {
  return (
    <span className={cx('Required', className)} {...props}>
      &#42;
    </span>
  );
}

export function FormElement({ className, ...props }) {
  return <div className={cx(`${s.FormElement}`, className)} {...props}></div>;
}

export function Textarea({ className, ...props }) {
  return <textarea className={cx(`${s.TextArea}`, className)} {...props} />;
}
