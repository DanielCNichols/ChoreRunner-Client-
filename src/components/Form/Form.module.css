@import '../../vars.css';

.FormElement {
  display: flex;
  flex-direction: column;
  text-align: left;
}

.Input,
.TextArea,
.Select {
  height: 35px;
  border-radius: 3px;
  padding: 3px;
}

.Label {
  font-size: 1.1em;
}

.Fieldset {
  border: none;
}

.Legend {
  font-family: var(--font-headers);
}
