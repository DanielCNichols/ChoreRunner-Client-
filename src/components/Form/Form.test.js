import React from 'react';
import ReactDOM from 'react-dom';
import { Input } from './Form';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Form component testing', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<MemoryRouter><Input /></MemoryRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('renders UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><Input /></MemoryRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Input />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
