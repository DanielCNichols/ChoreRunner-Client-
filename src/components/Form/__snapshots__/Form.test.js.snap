// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component testing renders UI as expected 1`] = `
<input
  className="Input"
  type="text"
/>
`;

exports[`Form component testing renders empty given no tabs 1`] = `
<input
  className="Input"
  type="text"
/>
`;
